### ISSUE TITLE
Size of generated files / IDE

### ISSUE THREAD

**Issue by Matt-Texier** [therecipe/qt#446]
Hi,

Nothing to do with anything regarding the software itself which is just great for golang lovers.

I am using golang IDE which help for productivity and in particular to parse object class and methods and correct issues on the fly. However, the size of files generated by qtsetup could be massive and makes the IDE suffering for indexing all class and methods.

Would it be possible in a future release to cut the biggest files into smaller slices (widgets.go, core.go, ...) ?

I would agree that it is not a key feature for sure ... but would be convenient.

Thanks Matt.
(PS: I did a super small PR on README file, looks like the CI process doesn't allocate enough resources to complete and as such PR failed ... just as FYI)

**Comment by therecipe** [comment]
Hey :)

>Would it be possible in a future release to cut the biggest files into smaller slices (widgets.go, core.go, ...) ?

Yeah, it should be possible but I'm not sure if this would solve the Gogland / IntelliJ issues.
(I will test it, if I find time though)

In the meantime you could try to increase the available memory size: https://github.com/therecipe/qt/wiki/FAQ#how-to-make-ides-code-completion-works-
That should make code completition for the widgets and gui package work.

>(PS: I did a super small PR on README file, looks like the CI process doesn't allocate enough resources to complete and as such PR failed ... just as FYI)

Merged it, and yeah the travis/circle ci/... builds are partially broken since a while.
But I will get rid of most of them in the next few days anyway, because the whole CI process was replaced with Vagrant a while ago. Thanks for the heads up though :)

**Comment by Matt-Texier** [comment]
Hey :) Thanks for your reply,

I just increased JVM memory of IntelliJ ... things are getting better but not as fast as it should be to be really usable. Anyway, not a big deal ...

Maybe a last question if I may ... I am working on a designing a software that will have to be in production by end of the year beginning of next year. 

Would you consider your qt golang wrapper mature enough for me to rely on it ?

I saw that you introduce a form of licensing for contributors ... Does it means that you plan a licensing at some point ?

Thanks Matthieu.

**Comment by therecipe** [comment]
>Would you consider your qt golang wrapper mature enough for me to rely on it ?

Mh, yeah it is actually mature enough for most small / medium use cases.

However, there could be still some API's missing that might be critical to have, such as those that are OpenGL related, or other lesser used functions such as: https://github.com/therecipe/qt/issues/416 

The other things that still need work are all related to the tooling, such a namespace issue related to qtmoc or some binary size issues related to qtminimal/qtmoc. (The first is crtical for medium / big projects, the second is just a bit annoying)

Beside these, there is also the need to cleanup the code generator as well as most of the other code, to make maintaining simpler as well as to make the tools more efficient, but that's has only low priority for now.

And lastly there is also the need to get proper versioning going + the project needs to be re-named.

But all in all the project is probably more mature than it looks from the outside, I suppose.

I also started working on a showcase recently to show how some medium sized QML project could look like (and how it can be structured). And I will also created a second one for the regular desktop widgets as well.
Maybe those will be helpful to evaluate the binding for your use cases.

>I saw that you introduce a form of licensing for contributors ... Does it means that you plan a licensing at some point ?

Yepp, there is the plan to use a dual license to commercialize the binding, but probably not before I was able to work on the issues I listed above.
